### PR TITLE
`Development`: Add documentation article for open-source

### PR DESCRIPTION
**Pull request by Mtze**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Documentation for open-source aspects of artemis are currently missing.
See [the confluence page](https://confluence.ase.in.tum.de/display/ArTEMiS/Feature+documentation).

### Steps for Testing

- Proof read the [new pages](https://artemis-platform--5372.org.readthedocs.build/en/5372/dev/open-source/)
- Deploy the PR to a test server and check that the new link is present on the About Us page

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by TheZoker — APPROVED** [review]
LGTM 👍

**Comment by bassner** [comment]
@Marvin-Vitcu That's because it links to this PR which isn't deployed to the production docs site yet, so it can't work and is fine

File: src/main/webapp/app/core/about-us/about-us.component.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class AboutUsComponent implements OnInit {
         ['scalable', {}],
         ['highUserSatisfaction', {}],
         ['customizable', {}],
-        ['openSource', {}],
+        ['openSource', { openSourceUrl: 'https://docs.artemis.ase.in.tum.de/dev/open-source.html' }],
```
Comment (bassner) [inline src/main/webapp/app/core/about-us/about-us.component.ts:45]: Same as above?
```suggestion
        ['openSource', { openSourceUrl: 'https://docs.artemis.ase.in.tum.de/dev/open-source/' }],
```

**Review by bassner — CHANGES_REQUESTED** [review]

**Review by Marvin-Vitcu — DISMISSED** [review]

**Comment by Mtze** [comment]
@bassner thanks for the suggestion!

File: docs/dev/open-source.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Open-Source
+===========
+
+We are dedicated to provide great software support for excellent teaching. To achieve that, we develop Artemis as `free and open-source <https://www.youtube.com/watch?v=EMi3PCW23yQ>`__ software. The core ideas of free software always have been "use", "study", "share", and "improve". 
+
+With Artemis, we embrace all of the aspects:
```
Comment (b-fein) [inline docs/dev/open-source.rst:6]:
```suggestion
With Artemis, we embrace all of those aspects:
```
Nipick: I think either ‘all the’ or ‘all of those’ sounds slightly better.

File: docs/dev/open-source.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+* Artemis is **used** by over ten universities and thousands of students. 
+* Artemis is part of active research and is constantly improved by students as part of university courses or masters/bachelors theses. **Studying** the Artemis source code helps our students to improve their software engineering skills while working on a real, actively used product.  
+* The foundation of learning and teaching is **sharing** knowledge. Artemis is used to conduct courses at universities and, at the same time, used as a education research platform.
```
Comment (b-fein) [inline docs/dev/open-source.rst:10]:
```suggestion
* The foundation of learning and teaching is **sharing** knowledge. Artemis is used to conduct courses at universities and, at the same time, used as an education research platform.
```

**Comment by Mtze** [comment]
Thanks @b-fein :)

**Review by b-fein — APPROVED** [review]

**Review by bassner — APPROVED** [review]

**Review by SimonPaul02 — APPROVED** [review]
Code looks good to me, tested on TS1, link works! (But it is not yet available)
image

**Review by yuling-sun — APPROVED** [review]
Tested on TS1. Link works as described.

**Review by jbllhff — APPROVED** [review]
Tested on ts2, looks good!

**Review by sleiss — APPROVED** [review]

**Review by luisa-we — APPROVED** [review]
Tested on TS1, the link works.
Documentation also looks good :)

**Comment by Mtze** [comment]
Thanks everyone :)

**Review by krusche — APPROVED** [review]
Sounds great 👍

**Pull request merged**